src/elements/content-sidebar/__tests__/withSidebarAnnotations.test.js

```diff
@@ -0,0 +1,267 @@
+
+    describe('addAnnotation()', () => {
+        beforeEach(() => {
+            annotatorContextProps.getAnnotationsMatchPath.mockReturnValueOnce({ params: { fileVersionId: '123' } });
+        });
+
+        test('should throw if no user', () => {
+            const instance = getWrapper({ annotatorState: { meta: { requestId: '123' } } }).instance();
+
+            expect(() => instance.addAnnotation()).toThrow('Bad box user!');
+            // One call to getAnnotationsMatchPath comes in the constructor
+            expect(annotatorContextProps.getAnnotationsMatchPath).toHaveBeenCalledTimes(2);
+        });
+
+        test('should do nothing if meta or requestId is not present', () => {
+            const instance = getWrapper().instance();
+
+            instance.addAnnotation();
+
+            // Only call to getAnnotationsMatchPath comes in the constructor, the one in addAnnotation should not occur
+            expect(annotatorContextProps.getAnnotationsMatchPath).toHaveBeenCalledTimes(1);
+        });
+
+        test.each`
+            title                                                             | isOpen   | isPending | isAnnotationsPath | hasItems | expectedAddCount | expectedRefreshCount
+            ${'add annotation and refresh sidebar for pending annotation'}    | ${true}  | ${true}   | ${true}           | ${true}  | ${1}             | ${1}
+            ${'refresh sidebar for pending annotation'}                       | ${false} | ${true}   | ${true}           | ${true}  | ${1}             | ${0}
+            ${'add annotation but not refresh if not annotation path'}        | ${true}  | ${true}   | ${false}          | ${true}  | ${1}             | ${0}
+            ${'not add annotation but refresh sidebar if cache has no items'} | ${true}  | ${true}   | ${true}           | ${false} | ${0}             | ${1}
+            ${'add annotation and refresh sidebar for completed annotation'}  | ${true}  | ${false}  | ${true}           | ${true}  | ${1}             | ${1}
+        `(
+            'should $title',
+            ({ isOpen, isPending, isAnnotationsPath, hasItems, expectedAddCount, expectedRefreshCount }) => {
+                const annotatorStateMock = {
+                    action: isPending ? 'create_start' : null,
+                    annotation: {},
+                    meta: {
+                        requestId: '123',
+                    },
+                };
+
+                const wrapper = getWrapper({ annotatorState: annotatorStateMock, currentUser, isOpen });
+                const instance = wrapper.instance();
+
+                annotatorContextProps.getAnnotationsMatchPath.mockReturnValueOnce(isAnnotationsPath);
+                feedAPI.getCachedItems.mockReturnValueOnce({ items: hasItems });
+                instance.sidebarPanels = {
+                    current: {
+                        refresh: jest.fn(),
+                    },
+                };
+
+                instance.addAnnotation();
```

Comment:
If we want to convert this to a function component later, we should consider how we'd test it if we didn't have direct access to the inner instance methods. Is there a way we can test the inputs/outputs while treating the logic like a black box?